react/features/base/dialog/components/Dialog.native.js

```diff
@@ -0,0 +1,127 @@
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+import { translate } from '../../i18n';
+
+import Prompt from 'react-native-prompt';
+
+import { hideDialog } from '../actions';
+
+/**
+ * Native dialog using Prompt.
+ */
+class Dialog extends Component {
+
+    /**
+     * Dialog component's property types.
+     *
+     * @static
+     */
+    static propTypes = {
+        cancelTitleKey: React.PropTypes.string,
+        dispatch: React.PropTypes.func,
+        onCancel: React.PropTypes.func,
+        onSubmit: React.PropTypes.func,
```

Comment:
This suggests dialogs will always have some submit and cancel buttons, what if we need more? I'd like to get an array ob objects. Each object represents a button, with its text + the associated handler function. WDYT?